api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -445,21 +447,50 @@ private DbUser updateDuccAgreement(
     return dbUser;
   }
 
+  /**
+   * This method is used by create Account flow, it throws exception if user has not accepted AoU
+   * Terms of Service at all or has accepted the incorrect version
+   *
+   * @param tosVersion
+   */
   @Override
-  public void validateTermsOfService(Integer tosVersion) {
+  public void validateAllOfUsTermsOfService(Integer tosVersion) {
     if (tosVersion == null) {
       throw new BadRequestException("Terms of Service version is NULL");
     }
     if (tosVersion != LATEST_AOU_TOS_VERSION) {
-      throw new BadRequestException("Terms of Service version is not up to date");
+      throw new BadRequestException("All of Us Terms of Service version is not up to date");
     }
   }
 
+  // Returns true if user has accepted the latest AoU Terms of Service Version
   @Override
-  public void validateTermsOfService(@Nonnull DbUser dbUser) {
+  public boolean validateAllOfUsTermsOfServiceVersion(@Nonnull DbUser dbUser) {
     final int tosVersion =
         userTermsOfServiceDao.findByUserIdOrThrow(dbUser.getUserId()).getTosVersion();
-    validateTermsOfService(tosVersion);
+    return tosVersion == LATEST_AOU_TOS_VERSION;
+  }
+
+  // Returns true only if the user has accepted the latest version of both AoU and Terra terms of
+  // service
+  @Override
+  public boolean validateTermsOfService(@Nonnull DbUser dbUser) {
+    boolean userHasAcceptedLatestTerraTos = getUserTerraTermsOfServiceStatus(dbUser);
+    return validateAllOfUsTermsOfServiceVersion(dbUser) && userHasAcceptedLatestTerraTos;
```

Comment:
If the AoU TOS is invalid, this version will make an unnecessary Terra call.  Short-circuit semantics mean that this version will not make the Terra call:

```suggestion
    return validateAllOfUsTermsOfServiceVersion(dbUser) && getUserTerraTermsOfServiceStatus(dbUser);
```